### PR TITLE
Find implicitNotFoundMessage on types

### LINKED ISSUE #11797: @implicitNotFound is ignored when the queried type is a type member (not a trait or a class)
## Compiler version

3.0.0-RC1

## Minimized code

```Scala
object Foo:
  @annotation.implicitNotFound("Oops")
  type Bar

@main def run(): Unit =
  summon[Foo.Bar]
```

https://scastie.scala-lang.org/KpsRCJFXRNOrDzfael6tbw

## Output

The compilation error is:

~~~ text
no implicit argument of type Foo.Bar was found for parameter x of method summon in object Predef
~~~

## Expectation

The compilation error message should display the content of the `@implicitNotFound` annotation. This is the case in Scala 2, FYI: https://scastie.scala-lang.org/YULABebLTOa66sTci8OLqg

However, there is a subtlety here. In the case of transparent type aliases, maybe we don’t want the compilation process to be impacted by the dealiasing process. But I see at least two good valid situations where we want the annotation to be taken into account: for `opaque` type aliases, and for abstract type members.

### PR DESCRIPTION
**Pull request by odersky**
Fixes #11797

### CODE REVIEW

File: tests/neg/i11797.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+object Foo:
+  @annotation.implicitNotFound("Oops")
+  type Bar
```
Comment (julienrf) [inline tests/neg/i11797.scala:3]: Do we want to also add a test with an `opaque type`? Are `opaque types` just represented as a `TypeRef` like transparent types?

**Review by julienrf — APPROVED** [review]

File: tests/neg/i11797.scala (hunk 1 above)
Comment (odersky) [reply tests/neg/i11797.scala:3]: yes that's the same

**Comment by adamgfraser** [comment]
Does this work when the type is just a type alias like this?

```scala
import scala.annotation.implicitNotFound

@implicitNotFound("${A} is not a subtype of ${B}")
type IsSubtypeOf[-A, +B] = A <:< B

object Example {
  implicitly[String IsSubtypeOf Int]
}
```

**Comment by odersky** [comment]
> Does this work when the type is just a type alias like this?

No, since the compiler is free to dealias the type before attempting an implicit search. It has to be an abstract type or opaque type.

**Comment by adamgfraser** [comment]
Would it be possible to support that? That functionality exists on Scala 2 and can be helpful for library authors to be able to alias existing types and provide more specific error messages associated with an implicit not being found rather than having to create new data types to wrap the existing ones.

**Comment by julienrf** [comment]
> It has to be an abstract type or opaque type.

How is this situation handled?

~~~ scala
trait Foo:
  @implicitNotFound("Oops")
  type Bar

trait Foo2 extends Foo:
  type Bar = Unit

object Foo3 extends Foo2:
  summon[Bar]
~~~

Although `Bar` is initially defined as an abstract type member in `Foo`, it becomes a transparent type alias in `Foo2`.

**Comment by odersky** [comment]
@julienrf Looks like this falls back to the generic case as well.

**Comment by odersky** [comment]
I had a look: it does not look easy to improve on this. It seems the type is dealiased way before we try to compose an error message.

**Comment by adamgfraser** [comment]
@odersky Thanks for looking!

**Pull request merged**